sds_data_manager/lambda_code/SDSCode/indexer.py

```diff
@@ -105,6 +107,49 @@ def _create_open_search_client():
     )
 
 
+def initialize_data_processing_status(filename: str):
+    """Generate data that will be sent to database.
+
+    Parameters
+    ----------
+    filename : str
+        filename of injested data.
+
+    Returns
+    -------
+    dict
+        data for database
+    """
+    # Get instrument name by parsing the filename.
+    # Eg. filename = imap_l0_sci_codice_20230602_v02.pkts
+    # Get instrument name, codice, from the filename.
+    instrument_name = filename.split("_")[3]
+
+    return {
+        "instrument": instrument_name,
+        "filename": filename,
+        "processing_status": ProcessingStatus.PENDING.name,
+        "l1a_status": ProcessingStatus.PENDING.name,
+        "l1b_status": ProcessingStatus.PENDING.name,
+        "l1c_status": ProcessingStatus.PENDING.name,
+        "l2_status": ProcessingStatus.PENDING.name,
+        "l3_status": ProcessingStatus.PENDING.name,
+    }
```

Comment:
I think I was interpreting this table as being "keep track of file status". So, I was just thinking that each file was only associated with one level. Meaning we could take out all of the other level fields and just store whichever single one we are on with this individual file. So, has this file not been created, currently being created, processing failed, etc... But all for only the status of this single file.

But, what I think you're describing is that each file may want to keep track of what the processing status is of the l1a/l1b before it, and the l3 after it that are all dependencies or dependent on this single file?

It might be good to lay out the schema here or link to the schema somewhere externally, just so we are explicitly noting what should go into this table and how a processing job may update it etc... I think I'm just missing the overall workflow. Happy to hop on a call and discuss if that is easier too rather than async via the review.

Reply:
I think this is a discussion we need to have as a team. I can create a subtask issue to discuss this.